website/static/js/pages/FileEditor.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+var ko = require('knockout');
+var $ = require('jquery');
+var $osf = require('js/osfHelpers');
+var Raven = require('raven-js');
+var Markdown = require('pagedown-ace-converter');
+Markdown.getSanitizingConverter = require('pagedown-ace-sanitizer').getSanitizingConverter;
+require('imports?Markdown=pagedown-ace-converter!pagedown-ace-editor');
+
+
+/**
+ * Binding handler that instantiates an ACE editor.
+ * The value accessor must be a ko.observable.
+ * Example: <div data-bind="ace: currentText" id="editor"></div>
+ */
+var editor;
+ko.bindingHandlers.ace = {
+    init: function (element, valueAccessor) {
+        editor = ace.edit(element.id);  // jshint ignore: line
+
+        // Updates the view model based on changes to the editor
+        editor.getSession().on('change', function () {
+            valueAccessor()(editor.getValue());
+        });
+    },
+    update: function (element, valueAccessor) {
+        var content = editor.getValue();        // Content of ace editor
+        var value = ko.unwrap(valueAccessor()); // Value from view model
+
+        // Updates the editor based on changes to the view model
+        if (value !== undefined && content !== value) {
+            var cursorPosition = editor.getCursorPosition();
+            editor.setValue(value);
+            editor.gotoLine(cursorPosition.row + 1, cursorPosition.column);
+        }
+    }
+};
+
+function ViewModel(url, viewText) {
+    var self = this;
+
+    self.initText = ko.observable('');
+    self.currentText = viewText; //from filePage's VM
+    self.activeUsers = ko.observableArray([]);
+    self.status = ko.observable('connecting');
+    self.throttledStatus = ko.observable(self.status());
```

Comment:
I'm a little confused about why throttledStatus is needed. Can you please explain?